lms/static/scripts/frontend_apps/services/grading.js

```diff
@@ -0,0 +1,66 @@
+import { apiCall } from '../utils/api';
+
+/**
+ * @typedef Student
+ * @prop {string} LISResultSourcedId - Unique outcome identifier
+ * @prop {string} LISOutcomeServiceUrl - API URL for posting outcome results
+ */
+
+/**
+ * The fetched grade success result.
+ *
+ * @typedef FetchGradeResult
+ * @prop {number} currentScore - The fetched grade
+ */
+
+/**
+ * Service for fetching and reporting student grades for an assignment.
+ */
+export class GradingService {
+  /**
+   * @param {object} options
+   *   @param {string} options.authToken
+   */
+  constructor({ authToken }) {
+    this._authToken = authToken;
+  }
+
+  /**
+   * Submits a student's grade to the LTI endpoint.
+   *
+   * @param {object} options
+   *   @param {Student} options.student - Student object
+   *   @param {number} options.grade - A number between 0 and 1
+   * @return {Promise<unknown>}
```

Comment:
Curious why the `Promise` resolution is `unknown`?

Reply:
`Promise<void>` would be better. The caller is not expected to use the result. I'll change it.